.pre-commit-config.yaml

```diff
@@ -50,5 +51,11 @@ repos:
   rev: v0.931
   hooks:
   - id: mypy
+    exclude: docs/.*
     additional_dependencies:
     - types-python-dateutil
+- repo: https://github.com/PyCQA/doc8.git
+  rev: 0.10.1
+  hooks:
+  - id: doc8
+    files: ^docs/
```

Comment:
I don't think we need doc8. I don't use it on any other projects.

Analyzing its rules:

* "invalid RST format - D000" - this will be caught by read the docs running build per PR.
* "lines should not be longer than 79 characters - D001" - I prefer to write in a one sentence per line style, I don't think it's useful to wrap at 79 chars as few people read docs in raw format.
* "no trailing whitespace - D002" - already handled by pre-commit's "trailing-whitespace"
* "no tabulation for indentation - D003" - handled by editorconfig
* " no carriage returns (use unix newlines) - D004" - handled by git which normalizes newlines on commit
* "no newline at end of file - D005" - handled by pre-commit's "end-of-file-fixer"